openedx/core/djangoapps/xmodule_django/models.py

```diff
@@ -159,6 +150,19 @@ def run_validators(self, value):
         return super(OpaqueKeyField, self).run_validators(value)
 
 
+class OpaqueKeyFieldEmptyLookupIsNull(IsNull):
+    """
+    This overrides the default __isnull model filter to help enforce the special way
+    we handle null / empty values in OpaqueKeyFields.
+    """
+    def get_prep_lookup(self):
+        raise TypeError("Use this field's .Empty member rather than None or __isnull "
+                        "to query for missing objects of this type.")
+
+
+OpaqueKeyField.register_lookup(OpaqueKeyFieldEmptyLookupIsNull)
```

Comment:
Any need to test this code to ensure that `TypeError` is raised in the right circumstances?
Or is it already tested somewhere?

Reply:
I tested it by hand on OpaqueKeyField subclasses in 1.8, but didn't see any specific unit tests around this (or any tests in xmodule_django).

Reply:
Hmm - isn't that special? I wish we hadn't implemented a custom field without tests directly on those fields. Oh well.